src/Interactive/Host/Interactive/Core/InteractiveHost.Service.cs

```diff
@@ -240,62 +257,37 @@ private static void RunServer(string serverPort, string semaphoreName, int clien
                     SetErrorMode(GetErrorMode() | ErrorMode.SEM_FAILCRITICALERRORS | ErrorMode.SEM_NOOPENFILEERRORBOX | ErrorMode.SEM_NOGPFAULTERRORBOX);
                 }
 
-                IpcServerChannel serverChannel = null;
-                IpcClientChannel clientChannel = null;
+                //(miziga) need try block with semaphore still?
                 try
                 {
-                    using (var semaphore = Semaphore.OpenExisting(semaphoreName))
-                    {
-                        // DEBUG: semaphore.WaitOne();
-
-                        var serverProvider = new BinaryServerFormatterSinkProvider();
-                        serverProvider.TypeFilterLevel = TypeFilterLevel.Full;
-
-                        var clientProvider = new BinaryClientFormatterSinkProvider();
 
-                        clientChannel = new IpcClientChannel(GenerateUniqueChannelLocalName(), clientProvider);
-                        ChannelServices.RegisterChannel(clientChannel, ensureSecurity: false);
-
-                        serverChannel = new IpcServerChannel(GenerateUniqueChannelLocalName(), serverPort, serverProvider);
-                        ChannelServices.RegisterChannel(serverChannel, ensureSecurity: false);
-
-                        RemotingConfiguration.RegisterWellKnownServiceType(
-                            typeof(Service),
-                            ServiceName,
-                            WellKnownObjectMode.Singleton);
-
-                        using (var resetEvent = new ManualResetEventSlim(false))
+                    NamedPipeServerStream serverStream = new NamedPipeServerStream(GenerateUniqueChannelLocalName(), PipeDirection.InOut, NamedPipeServerStream.MaxAllowedServerInstances, PipeTransmissionMode.Byte, PipeOptions.Asynchronous);
```

Comment:
>GenerateUniqueChannelLocalName() [](start = 83, length = 32)

This is not right.

Reply:
Is the issue with it taking in the pipename as a parameter?

Reply:
yup